### PR TITLE
updated manual add students

### LINKED ISSUE #29: Improve manual add students.
- Update manual add student
  - Prof can input the new students question if recieved by email
  - New button called "add and help" that immediatly adds that student to the queue and idicates they are being helped

Image
***Current manual add student to queue***

### PR DESCRIPTION
**Pull request by Ferdinand737**
# Description

Updated manual add students functionalit.
Professors can now add a question to each added student
Professors can now click a checkbox to immediatly start helping a student after adding them to the queue

Closes #29

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manually tested front-end code. No new integration test required because no back-end changes were made.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/app/components/Queue/TA/TAAddStudent.tsx [hunk 1]
```diff
@@ -197,6 +226,34 @@ export function AddStudentsModal({
             ) : (
               <p>There are no students or all students are in queue</p>
             )}
+            {selectOptions.length >= 1 ? (
```
Comment (d3li0n) [inline packages/app/components/Queue/TA/TAAddStudent.tsx:229]: `selectOptions.length > 0 ?` ?

File: packages/app/components/Queue/TA/TAAddStudent.tsx [hunk 2]
```diff
@@ -197,6 +226,34 @@ export function AddStudentsModal({
...
+                {selectOptions.map((option, index) => (
+                  <div key={index}>
+                    <strong>
+                      <p>{option.value}</p>
+                    </strong>
+                    <Form onFinish={handleSubmit}>
+                      <Form.Item>
+                        <input
+                          placeholder={`Enter ${option.value}'s question`}
+                          style={{ width: "100%", height: "40px" }}
```
Comment (d3li0n) [inline packages/app/components/Queue/TA/TAAddStudent.tsx:241]: I think you can just set `height: 40`?

File: packages/app/components/Queue/TA/TAAddStudent.tsx [hunk 3]
```diff
@@ -197,6 +226,34 @@ export function AddStudentsModal({
...
             )}
+            {selectOptions.length >= 1 ? (
+              <>
+                <br />
+                {selectOptions.map((option, index) => (
+                  <div key={index}>
+                    <strong>
+                      <p>{option.value}</p>
+                    </strong>
+                    <Form onFinish={handleSubmit}>
```
Comment (d3li0n) [inline packages/app/components/Queue/TA/TAAddStudent.tsx:237]: Are you creating multiple forms on purpose?

**Review by d3li0n — CHANGES_REQUESTED** [review]
Could you also show how it looks?

File: packages/app/components/Queue/TA/TAAddStudent.tsx (hunk 2 above)
Comment (Ferdinand737) [reply packages/app/components/Queue/TA/TAAddStudent.tsx:241]: `height: 40` does not work

File: packages/app/components/Queue/TA/TAAddStudent.tsx [hunk 4]
```diff
@@ -48,15 +49,37 @@ export function AddStudentsModal({
     "Default grading type"
   );
   const [selectOptions, setSelectOptions] = useState([]);
+
+  const handleCheckboxChange = (id) => {
```
Comment (d3li0n) [inline packages/app/components/Queue/TA/TAAddStudent.tsx:53]: I would probably leave comments explaining what this does and final structure of selectOptions look like. It's a bit hard to follow what's happening here

File: packages/app/components/Queue/TA/TAAddStudent.tsx [hunk 5]
```diff
@@ -91,21 +113,25 @@ export function AddStudentsModal({
...
         currentStudent.id
       )
-      .then(() => {
+      .then(async (response) => {
         message.success("Student(s) added");
         setStudentsState(
           studentsState.filter((student) => student.id !== currentStudent.id)
         );
         setSelectOptions([]);
+        if (selectOptions[i].help == true) {
```
Comment (d3li0n) [inline packages/app/components/Queue/TA/TAAddStudent.tsx:130]: Where are we getting `i` from?

File: packages/app/components/Queue/TA/TAAddStudent.tsx [hunk 6]
```diff
@@ -197,6 +238,35 @@ export function AddStudentsModal({
...
+                  {selectOptions.map((option, index) => (
+                    <div key={index}>
+                      <strong>
+                        <p>{option.value}</p>
+                      </strong>
+                      <Form.Item>
+                        <input
+                          placeholder={`Enter ${option.value}'s question`}
+                          style={{ width: "100%", height: "40px" }}
+                          onChange={(e) => handleChange(e, option.id)}
```
Comment (d3li0n) [inline packages/app/components/Queue/TA/TAAddStudent.tsx:254]: If we only need changed value, why not pass only value e.g. `e.target.value` instead of the whole object?

**Review by d3li0n — CHANGES_REQUESTED** [review]
Left some comments to consider

**Review by wskksw — APPROVED** [review]

**Comment by wskksw** [comment]
The code should work, and I approved the PR. 
While it's usable, the input (checkbox) and test UI seem not optimized. If you have extra time, you might consider add styles to it for usability.

File: packages/app/components/Queue/TA/TAAddStudent.tsx (hunk 5 above)
Comment (Ferdinand737) [reply packages/app/components/Queue/TA/TAAddStudent.tsx:130]: `addStudent` is in a loop
```
const handleSubmit = () => {
    selectOptions.forEach((student, i) => {
      addStudent(i);
    });
  };

  const addStudent = async (i) => {
    const currentStudent = selectOptions[i];
    const b = await API.profile.inQueue(currentStudent.id);
    if (b) {
      message.error("Student already in queue.");
      return;
    }
    await API.questions
      .TAcreate(
        {
          text: currentStudent.question ?? "",
          queueId: queueId,
          location: null,
          force: true,
          groupable: false,
        },
        currentStudent.id
      )
      .then(async (response) => {
        message.success("Student(s) added");
        setStudentsState(
          studentsState.filter((student) => student.id !== currentStudent.id)
        );
        setSelectOptions([]);
        if (selectOptions[i].help == true) {
          await API.questions.update(response.id, {
            status: OpenQuestionStatus.Helping,
          });
        }
      })
      .catch(() => {
        message.error("Can't add student".concat(currentStudent.value));
      });
    return false;
  };
```

**Review by d3li0n — APPROVED** [review]
